modules/mongodb/src/main/java/org/testcontainers/containers/MongoDBContainer.java

```diff
@@ -29,25 +30,54 @@ public class MongoDBContainer extends GenericContainer<MongoDBContainer> {
 
     private static final String MONGODB_DATABASE_NAME_DEFAULT = "test";
 
+    private static final String STARTER_SCRIPT = "/testcontainers_start.sh";
+
+    private final boolean shardingEnabled;
+
     /**
      * @deprecated use {@link MongoDBContainer(DockerImageName)} instead
      */
     @Deprecated
     public MongoDBContainer() {
-        this(DEFAULT_IMAGE_NAME.withTag(DEFAULT_TAG));
+        this(DEFAULT_IMAGE_NAME.withTag(DEFAULT_TAG), false);
```

Comment:
The following method can be added in order to opt-in for Sharding

```java
public MongoDBContainer withSharding() {
    this.shardingEnabled = true;
    return self();
}
```

```suggestion
        this(DEFAULT_IMAGE_NAME.withTag(DEFAULT_TAG));
```

Reply:
I had it this wait to start but the way the contructor currently works, it never sees that field.  Now that I have a working form and understand the lifecycle a bit more, I think I know how to disable the default script the container runs at start up and then run the sharding script in `containerStarted()`.

In fact, when looking back over this PR (in reviewer mode vs author mode) I realized this current set up may actually end up with a non-replSet config.  I'll confirm as I rework that constructor.

Reply:
There is another comment below which adds `configure` method. That way, the proper wait strategy is executed.